lib/foreman_openscap/helper.rb

```diff
@@ -16,25 +16,21 @@ def self.get_asset(cname, policy_id)
   end
 
   def self.find_name_or_uuid_by_host(host)
-    if defined?(::Katello::System)
-      host.content_host.uuid
-    else
-      host.name
-    end
+    host.respond_to?(:subscription_facet) ? host.subscription_facet.uuid : host.name
   end
 
   private
 
   def self.find_host_by_name_or_uuid(cname)
-    if defined?(Katello::System)
-      host = Host.includes(:content_host).where(:katello_systems => {:uuid => cname}).first
+    if defined?(Katello::Host::SubscriptionFacet)
```

Comment:
I would think the facets API should be the 'best' way to check for the existence of a facet rather than relying on a class that could change.